drv/transceivers-server/src/udp.rs

```diff
@@ -550,7 +643,15 @@ impl ServerImpl {
             HostRequest::ClearPowerFault(modules) => {
                 ringbuf_entry!(Trace::ClearPowerFault(modules));
                 let mask = LogicalPortMask::from(modules) & !self.disabled;
-                let result = self.transceivers.clear_power_fault(mask);
+
+                let result =
+                    if self.front_io_board_present == FrontIOStatus::Ready {
+                        self.transceivers.clear_power_fault(mask)
+                    } else {
+                        ModuleResultNoFailure::new(LogicalPortMask(0), mask)
+                            .unwrap_lite()
+                    };
```

Comment:
This is totally fine and functional. If you'd want to improve the readability of the
```
if self.front_io_board_present == FrontIOStatus::Ready {
    ... no front IO board error response
} else {
    ... actual operation
}
```
in every handler function a little bit you could consider making a static, pre canned `MessageBody::SpResponse(..)` for this case and change the handlers to
```
if self.front_io_board_present == FrontIOStatus::Ready {
    return sp_response_no_front_io_board;
}

... actual operation
```
That would remove this common check from the parse tree for anyone reading this code.

Reply:
I DM'd Arjen a bit about this and I'm not going to make any changes here. Given every request type has a corresponding response type (e.g. `HostRequest::Status` -> `SpResponse::Status`) there's really not a nice way to clean this up. I could make a separate code path that checks `front_io_board_present` prior to dropping into the message handler, but that just duplicats the same match structure elsewhere.